web.js

```diff
@@ -24,18 +25,17 @@ module.exports = function web (options) {
   var seneca = this
   var extend = seneca.util.deepextend
 
-  // Avoid deepextending context and auth,
-  // they aren't stringify friendly
-  opts = extend(opts, _.omit(options, ['context', 'auth']))
+  // Avoid deepextending context, middleware and auth, they aren't stringify friendly
+  opts = extend(opts, _.omit(options, ['context', 'auth', 'middleware']))
   opts.context = options.context || null
   opts.auth = options.auth || null
+  opts.middleware = options.middleware || null
```

Comment:
Change this to something like....

```js
if (options.middleware) {
  opts.options.middleware = options.middleware
}
```

this would allow for shorthand providing options at the root object instead of inside `options`.